src/main/java/org/folio/rest/impl/PurchaseOrdersAPI.java

```diff
@@ -39,16 +50,42 @@ public void getOrdersStoragePurchaseOrders(String query, int offset, int limit,
 
   @Override
   @Validate
-  public void postOrdersStoragePurchaseOrders(String lang, org.folio.rest.jaxrs.model.PurchaseOrder entity,
-      Map<String, String> okapiHeaders,
-      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    PgUtil.post(PURCHASE_ORDER_TABLE, entity, okapiHeaders, vertxContext, PostOrdersStoragePurchaseOrdersResponse.class, asyncResultHandler);
+  public void postOrdersStoragePurchaseOrders(String lang, org.folio.rest.jaxrs.model.PurchaseOrder entity, Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    try {
+      vertxContext.runOnContext(v -> {
+        Tx<PurchaseOrder> tx = new Tx<>(entity);
+        Future.succeededFuture(tx)
+          .compose(this::startTx)
+          .compose(this::createSequence)
+          .compose(t -> createPurchaseOrder(tx, entity, okapiHeaders, vertxContext))
```

Comment:
The `Tx` holds already `entity` so makes sense to remove second param from `createPurchaseOrder`